test/e2e/user_configured_cabundle_test.go

```diff
@@ -0,0 +1,692 @@
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: caConfigMapName,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// buildService returns service object for the provided port.
+func buildService(name types.NamespacedName, labels map[string]string, port int32) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name.Name,
+			Namespace: name.Namespace,
+			Labels:    labels,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{
+				Protocol: corev1.ProtocolTCP,
+				Port:     port,
+				TargetPort: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: port,
+				},
+			}},
+			Selector: labels,
+			Type:     corev1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+// buildReEncryptRoute returns route object for the provided service.
+func buildReEncryptRoute(name types.NamespacedName, svcName string, labels map[string]string, port int32, path string) *routev1.Route {
+	return &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name.Name,
+			Namespace: name.Namespace,
+			Labels:    labels,
+		},
+		Spec: routev1.RouteSpec{
+			Path: path,
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: port,
+				},
+			},
+			TLS: &routev1.TLSConfig{
+				Termination: routev1.TLSTerminationReencrypt,
+			},
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: svcName,
+			},
+		},
+	}
+}
+
+// getRouteHostName returns the host name in the route object.
+func getRouteHostName(ctx context.Context, name types.NamespacedName) (string, error) {
+	route := &routev1.Route{}
+	if err := kclient.Get(ctx, name, route); err != nil {
+		return "", err
+	}
+	return route.Spec.Host, nil
+}
+
+func getHTTPClient(cert, key, cacert string) (*http.Client, error) {
+	// load tls certificates
+	clientTLSCert, err := tls.X509KeyPair([]byte(cert), []byte(key))
+	if err != nil {
+		return nil, fmt.Errorf("failed to load client key pair: %v", err)
+	}
+
+	// Configure the client to trust TLS server certs issued by a CA.
+	certPool := x509.NewCertPool()
+	if ok := certPool.AppendCertsFromPEM([]byte(cacert)); !ok {
+		return nil, fmt.Errorf("failed to add CA certificate to pool")
+	}
+
+	tlsConfig := &tls.Config{
+		RootCAs:            certPool,
+		Certificates:       []tls.Certificate{clientTLSCert},
+		InsecureSkipVerify: true,
+	}
+	tr := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+	c := &http.Client{
+		Timeout:   1 * time.Minute,
+		Transport: tr,
+	}
+	return c, nil
+}
+
+// queryServer is for querying the service through the route created for the same as an
+// external client.
+func queryServer(t *testing.T, httpClient *http.Client, host, path string, expectedStatusCode int) (int, error) {
+	t.Helper()
+
+	const (
+		retries       = 3
+		retryInterval = time.Minute
```

Comment:
This might just be my opinion, but a minute quite high to sleep for between requests. Maybe 10 or 20 seconds at most?

I also think it doesn't hurt to do more retries, maybe 5 or 10. Since this is going through the internet, we deal with flakes, especially flakey DNS lookup issue within our CI cluster. Might seem overly generous, but flakes are quite time and resource consuming.

Reply:
Initially used a minute retry interval thinking of the time taken configmap changes to reflect in pod.
I have updated the retries and interval.